packages/rum/src/domain/deflate/deflateWorker.ts

```diff
@@ -61,10 +71,11 @@ export function startDeflateWorker(
   switch (state.status) {
     case DeflateWorkerStatus.Loading:
       state.initializationFailureCallbacks.push(onInitializationFailure)
-      return state.worker
+      return { worker: state.worker, stop: state.stop }
     case DeflateWorkerStatus.Initialized:
-      return state.worker
+      return { worker: state.worker, stop: state.stop }
   }
+  return { worker: undefined, stop: noop }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Instead of returning `stop` here, call `stop` in `resetDeflateWorkerState`. The DeflateWorker is a singleton, and if a test stops it without reseting its state, subsequent tests may fail.